src/main/java/org/candlepin/service/model/EventBody.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.model;
+
+import java.util.Objects;
+
+/**
+ * The event body contains the serialized text for the {@link Event}'s body and what type of serialization
+ * was used to create the text.
+ */
+public class EventBody {
+
+    private String body;
+    private SerializationType serializationType;
+
+    /**
+     * Creates a new event body with the provided body and serialization type.
+     *
+     * @param body
+     *  the serialized body of an {@link Event}
+     *
+     * @param serializationType
+     *  the type of serialization used to create the body text
+     *
+     * @throws IllegalArgumentException
+     *  if the provided body text is null or blank
+     *
+     * @throws NullPointerException
+     *  if the provided serialization type is null
+     */
+    public EventBody(String body, SerializationType serializationType) {
+        if (body == null) {
+            throw new IllegalArgumentException("event body is null");
+        }
+
+        if (body.isBlank()) {
+            throw new IllegalArgumentException("event body is blank");
+        }
+
+        this.body = body;
+        this.serializationType = Objects.requireNonNull(serializationType);
```

Comment:
Is there a reason that this method uses IAEs for body, but NPEs for serialization type?

Reply:
Good point, we should be consistent here. Updated the null check on serializationType to return an IAE instead of NPE and updated the javadoc.